beam/www/beam/pages/Move.vue

```diff
@@ -8,36 +7,81 @@
 			<RouterLink :to="{ name: 'home' }">Home</RouterLink>
 		</template>
 	</Navbar>
+	<div class="move">
+		<div class="dropdown-container">
+			<ADropdown label="Source Warehouse" :items="warehouseList" v-model="stockEntry.from_warehouse" />
+			<BeamBtn class="clear-button" @click="clearField('from_warehouse')"> X </BeamBtn>
+		</div>
+		<div class="dropdown-container">
+			<ADropdown label="Target Warehouse" :items="warehouseList" v-model="stockEntry.to_warehouse" />
+			<BeamBtn class="clear-button" @click="clearField('to_warehouse')"> X </BeamBtn>
+		</div>
+	</div>
 
 	<!-- body section -->
-	<ListView :items="items" :key="componentKey" />
+	<ListView :items="items" :key="componentKey" @update="update" />
 	<div class="begin" v-if="items.length == 0">
-		<span>Scan to Begin</span>
+		<span>Scan Items and Scan or Select Warehouses to Begin</span>
 	</div>
-
 	<!-- footer section -->
-	<ControlButtons :buttons="controlButtons" />
+	<ControlButtons :buttons="controlButtons"  />
 </template>
 
 <script setup lang="ts">
 import type { ListViewItem } from '@stonecrop/beam'
 import { ref, onMounted, computed } from 'vue'
 
 import ControlButtons from '@/components/ControlButtons.vue'
+import ADropdown from '@/components/ADropdown.vue'
 import { useBeamStore } from '@/stores/beam'
-import type { ControlButton, StockEntry } from '@/types'
+import type { ControlButton, DocActionResponse, StockEntry } from '@/types'
+import { watch } from 'vue'
+type Warehouse = {
+	name: string
+}
 
 const store = useBeamStore()
-
 const items = ref<ListViewItem[]>([])
-const stockEntryId = ref('')
-const stockEntry = ref(store.cache.mappers[stockEntryId.value] as StockEntry)
+const stockEntry = computed(
+	(): StockEntry =>
+		(store.cache.mappers[''] as StockEntry) || {
+			name: '',
+			stock_entry_type: 'Material Transfer',
+			items: [],
+			from_warehouse: '',
+			to_warehouse: '',
+		}
+)
 const componentKey = ref(0)
 
+const warehouseList = ref<string[]>([])
+
 onMounted(async () => {
 	store.form as Partial<StockEntry>
+	store.$patch(state => {
+		state.cache.mappers[''] = stockEntry.value
+	})
+	await loadWarehouses()
 })
 
+const loadWarehouses = async () => {
+	const warehouses = await store.getAll<Warehouse[]>('Warehouse', {
+		filters: JSON.stringify([['is_group', '!=', '1']]),
+	})
+	warehouseList.value = warehouses.map(warehouse => warehouse.name)
+}
+
+const clearField = (field: 'from_warehouse' | 'to_warehouse') => {
+	store.$patch(state => {
+		const mapper = state.cache.mappers['']
+		if (mapper) mapper[field] = ''
+	})
+}
```

Comment:
@lauty95 This should probably just update the local reference to the Stock Entry object, and skip the mapper